examples/api/test/widgets/scroll_view/custom_scroll_view.1_test.dart

```diff
@@ -0,0 +1,244 @@
+// Copyright 2014 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'package:flutter/gestures.dart';
+import 'package:flutter/material.dart';
+import 'package:flutter_api_samples/widgets/scroll_view/custom_scroll_view.1.dart' as example;
+import 'package:flutter_test/flutter_test.dart';
+
+void main() {
+  testWidgets('Widget tree is visible', (WidgetTester tester) async {
+    await tester.pumpWidget(const example.CustomScrollViewExampleApp());
+
+    // The crucial Widgets are:
+    // - Scaffold
+    //    - Appbar
+    //      - IconButton
+    //    - CustomScrollView
+    //       - SliverList (top, initially not existing)
+    //       - SliverList (bottom, with one element)
+
+    expect(
+      find.byType(Scaffold),
+      findsOne,
+      reason: 'Expected to have a Scaffold in the App',
+    );
+
+    expect(
+      find.descendant(
+        of: find.byType(Scaffold),
+        matching: find.byType(AppBar),
+      ),
+      findsOne,
+      reason: 'Expected to have an Appbar in the Scaffold',
+    );
+
+    expect(
+      find.descendant(
+        of: find.byType(AppBar),
+        matching: find.byType(IconButton),
+      ),
+      findsOne,
+      reason: 'Expected an IconButton in the AppBar',
+    );
+
+    expect(
+      find.descendant(
+        of: find.byType(Scaffold),
+        matching: find.byType(CustomScrollView),
+      ),
+      findsOne,
+      reason: 'Expected a CustomScrollView in the Scaffold',
+    );
+
+    // Initially, there is only one SliverList
+    expect(
+      find.descendant(
+        of: find.byType(CustomScrollView),
+        matching: find.byType(SliverList),
+      ),
+      findsOne,
+      reason: 'Expected one SliverList in the CustomScrollView',
+    );
+
+    //CustomScrollView contains one element 'Item: 0'
+    expect(
+      find.descendant(
+        of: find.byType(CustomScrollView),
+        matching: find.text('Item: 0'),
+      ),
+      findsOne,
+      reason: 'Expected element with text "Item: 0" in the CustomScrollView',
+    );
+  });
+  testWidgets('IconButton click extends existing SliverList',
+      (WidgetTester tester) async {
+    await tester.pumpWidget(const example.CustomScrollViewExampleApp());
+
+    // Initially, there is only one SliverList in the CustomScrollView
+    expect(
+      find.descendant(
+        of: find.byType(CustomScrollView),
+        matching: find.byType(SliverList),
+      ),
+      findsOne,
+      reason:
+          'Expected one, initial (bottom), SliverList in the CustomScrollView',
```

Comment:
Thanks for adding in the additional `reason`, I don't see that being done very often!

I would recommend getting rid of the line break (here and below), based on one of the [style guidelines](https://github.com/flutter/flutter/blob/05f294c7116f30a93195b479d55539f457edd83f/docs/contributing/Style-guide-for-Flutter-repo.md#prefer-a-maximum-line-length-of-80-characters):

```suggestion
      reason: 'Expected one, initial (bottom), SliverList in the CustomScrollView',
```